src/d/actor/d_a_obj_pfall.cpp

```diff
@@ -5,146 +5,473 @@
+    mDoMtx_stack_c::ZrotM(field_0x3B4);
+    mDoMtx_stack_c::transM(0.0f, 10.0f, 0.0f);
+    field_0x3C4->setBaseTRMtx(mDoMtx_stack_c::get());
+    cMtx_copy(mDoMtx_stack_c::get(), field_0x3CC);
+
+    field_0x3FC->setBaseScale(scale);
+    mDoMtx_stack_c::transS(current.pos);
+    mDoMtx_stack_c::YrotM(current.angle.y);
+    mDoMtx_stack_c::transM(offset_pos_right);
+    mDoMtx_stack_c::YrotM( -0x8000);
+    mDoMtx_stack_c::transM(0.0f, -10.0f, 0.0f);
+    mDoMtx_stack_c::ZrotM(field_0x3B4);
+    mDoMtx_stack_c::transM(0.0f, 10.0f, 0.0f);
+    field_0x3FC->setBaseTRMtx(mDoMtx_stack_c::get());
+    cMtx_copy(mDoMtx_stack_c::get(), field_0x404);
+
+    J3DModel* pModel = mpMorf->getModel();
+    static cXyz offset_pos(0.0f, 0.0f, -130.0f);
+    static cXyz scale(0.5f, 0.5f, 0.5f);
+
+    pModel->setBaseScale(scale);
+    mDoMtx_stack_c::transS(current.pos);
+    mDoMtx_stack_c::YrotM(current.angle.y);
+
+    mDoMtx_stack_c::transM(offset_pos);
+    pModel->setBaseTRMtx(mDoMtx_stack_c::get());
+    mDoMtx_stack_c::transM(0.0f, field_0x438 + l_HIO.field_0x08, 0.0f);
+    mDoMtx_stack_c::YrotM(-0x8000);
+    mpHimoModel->setBaseTRMtx(mDoMtx_stack_c::get());
 }
 
 /* 00000518-00000858       .text CreateHeap__13daObj_Pfall_cFv */
-void daObj_Pfall_c::CreateHeap() {
-    /* Nonmatching */
+BOOL daObj_Pfall_c::CreateHeap() {
+    J3DModelData* modelData = (J3DModelData*)dComIfG_getObjectRes("Pfall", PFALL_BDL_AOTSI);
+    JUT_ASSERT(0xD1, modelData != NULL);
+    field_0x3C4 = mDoExt_J3DModel__create(modelData, 0, 0x11020203);
+    field_0x3FC = mDoExt_J3DModel__create(modelData, 0, 0x11020203);
+
+    if(field_0x3C4 == NULL || field_0x3FC == NULL) {
+        return FALSE;
+    }
+
+    J3DModelData* himoModelData = (J3DModelData*)dComIfG_getObjectRes("Pfall", PFALL_BDL_PSUZU);
+    JUT_ASSERT(0xD9, himoModelData != NULL);
+    mpHimoModel = mDoExt_J3DModel__create(himoModelData, 0, 0x11020203);
+
+    if (mpHimoModel == NULL) {
+        return FALSE;  
+    }
+
+    J3DModelData* nzModelData = (J3DModelData*)dComIfG_getObjectRes("Pfall", PFALL_BDL_NZ);
+    JUT_ASSERT(0xE0, nzModelData != NULL);
+    mpMorf = new mDoExt_McaMorf(
+        nzModelData,
+        NULL, NULL,
+        NULL,
+        -1, 1.0f, 0, -1, 1,
+        NULL,
+        0x00080000,
+        0x11000002
+    );
+
+    if(mpMorf == NULL || mpMorf->getModel() == NULL) {
+        return FALSE;
+    }
+    mpMorf->getModel()->setUserArea((u32)this);
+    set_mtx();
+    mpMorf->calc();
+    mpBgW = new dBgW();
+
+    if(mpBgW == NULL) {
+        return FALSE;
+    }
+    cBgD_t* pData = (cBgD_t*)dComIfG_getObjectRes("Pfall", PFALL_DZB_AOTOSI);
+
+    if (mpBgW->Set(pData, cBgW::MOVE_BG_e, &field_0x3CC) == true) {
+        return FALSE;
+    }
+    mpBgW2 = new dBgW();
+
+    if(mpBgW2 == NULL) {
+        return FALSE;
+    }
+    
+    pData = (cBgD_t*)dComIfG_getObjectRes("Pfall", PFALL_DZB_AOTOSI);
+    
+    return (mpBgW2->Set(pData, cBgW::MOVE_BG_e, &field_0x404) & 0xff) != 1 ? 1 : 0;
```

Comment:
Does this still match?
```suggestion
    if (mpBgW2->Set(pData, cBgW::MOVE_BG_e, &field_0x404) == true) {
        return FALSE;
    }

    return TRUE;
```